lib/spack/spack/build_systems/python.py

```diff
@@ -128,191 +112,79 @@ def import_modules(self):
             list: list of strings of module names
         """
         modules = []
-        root = os.path.join(
-            self.prefix,
-            self.spec['python'].package.config_vars['python_lib']['true']['false'],
-        )
 
-        # Some Python libraries are packages: collections of modules
-        # distributed in directories containing __init__.py files
-        for path in find(root, '__init__.py', recursive=True):
-            modules.append(path.replace(root + os.sep, '', 1).replace(
-                os.sep + '__init__.py', '').replace('/', '.'))
-
-        # Some Python libraries are modules: individual *.py files
-        # found in the site-packages directory
-        for path in find(root, '*.py', recursive=False):
-            modules.append(path.replace(root + os.sep, '', 1).replace(
-                '.py', '').replace('/', '.'))
+        # Packages may be installed in platform-specific or platform-independent
+        # site-packages directories
+        config_vars = self.spec['python'].package.config_vars
+        site_packages_dirs = set()
+        for plat_specific in ['true', 'false']:
+            try:
+                site_packages_dirs.add(
+                    config_vars['python_lib'][plat_specific]['false']
+                )
+            except KeyError:
+                site_packages_dirs.add(self.default_site_packages_dir)
+
+        for site_packages_dir in site_packages_dirs:
```

Comment:
Why this rather than `--target` and using the original single-directory logic?

Reply:
The original single-directory logic does not work. There never was (and likely never will be) a single `site-packages` directory for Python installations. For Spack-installed Python, purelib and platlib are the same thing. For externally installed Python, purelib and platlib are often two different directories. We can force all Spack-installed Python libraries to only install to a single prefix, but if we want to support externally-installed Python (and externally-installed Python libraries) we need to support both. If we don't force all Spack-installed Python libraries to use the same site-packages directory, some packages get installed in purelib and some get installed in platlib. So I think our best option is to not rock the boat too much and just let pip/setuptools do their thing and load both. See https://discuss.python.org/t/understanding-site-packages-directories/12959 for more discussion.

Reply:
Externals are a good reason for that I suppose.  Seemed extraneous from a spack installed perspective but if it keeps duplication down for externals I'm all good.

Reply:
Yep, 90% of the hacks in our Python package are for supporting externals. Reading https://ffy00.github.io/blog/02-python-debian-and-the-install-locations/ may make you cry lol.